api/server/server.go

```diff
@@ -1340,34 +1338,29 @@ func postContainerExecCreate(eng *engine.Engine, version version.Version, w http
 		return nil
 	}
 	var (
-		name         = vars["name"]
-		job          = eng.Job("execCreate", name)
-		stdoutBuffer = bytes.NewBuffer(nil)
-		outWarnings  []string
-		warnings     = bytes.NewBuffer(nil)
+		name = vars["name"]
 	)
 
-	if err := job.DecodeEnv(r.Body); err != nil {
+	execConfig := runconfig.ExecConfig{}
```

Comment:
In ExecConfig there's a "container" field but the old code didn't decoded a value into it.  But, this new code will allow a client to set it if they include "container" in the incoming json. I looked and I didn't see any code in d.ContainerExecCreate() using that field so I think its safe but @LK4D4 can you verify there isn't an issue here (in particular a security one) ?